File: examples/typescript/react/notes-showcase/src/models/notebookModel.ts
/**
 * NotebookModel — pure-data record for a notebook node.
 *
 * Immutable (readonly fields). `parentId` is null for root notebooks.
 * Identifiers are stable strings — same as C# / Python flavors so
 * cross-language parity audits compare identically.
 *
 * `isReadonly` (default `false`) marks a notebook whose notes cannot be
 * created or edited via the UI. When the currently-bound notebook is
 * readonly `CapabilityActionsVM` disables the *Add Note* action.
 */
export interface NotebookModel {
  readonly id: string;
  readonly name: string;
  readonly parentId: string | null;
  readonly isReadonly?: boolean;
}

File: examples/python/textual/notes_showcase/src/notes_showcase/models/notebook_model.py
"""NotebookModel — pure-data record for a notebook node."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class NotebookModel:
    """Immutable notebook record.

    ``parent_id`` is ``None`` for root notebooks. Identifiers are stable
    strings (mirrors the C# / TS flavors so cross-language parity audits
    compare identically).

    ``is_readonly`` (default ``False``) marks a notebook whose notes cannot
    be created or edited via the UI. When the currently-bound notebook is
    readonly :class:`CapabilityActionsVM` disables the *Add Note* action.
    """

    id: str
    name: str
    parent_id: str | None
    is_readonly: bool = False
